### ISSUE TITLE
Bounding the velocity to laneMaxV

### ISSUE THREAD

**Issue by Domsall** [eclipse/sumo#7644]
Hello,
In MSVehicle.cpp in planMoveInterval the velocity v is the initial maximum velocity and bounded by laneMaxV:
`double v = MIN2(maxV, laneMaxV);`
Therefore, dawdling over the vehicle specific laneMaxV is not possible in this part of the calculation and is added in the CF-model function finalizeSpeed. I integrated dawdling/imperfect estimation values in stopSpeed/followSpeed, which results in dawdling around the driven speed. But this this kind of implementation, v+dawdling can't become higher than laneMaxV (when the vehicle drives near laneMaxV).
Is it possible to change the initial maximum velocity to `double v = maxV;`? A few tests of mine showed, that the velocity is bounded elsewhere and often in the CF-model anyway, which is also where I would expect the bound to happen, as the different CF-models would maybe use different boundaries. Nevertheless, I don't have the comprehensive understanding of all dependecies and have not carried out a full testset yet, so I hope somebody could help and clarify if this makes sense.
Thanks in advance!

**Comment by namdre** [comment]
I don't have time to test this at the moment but here are some possible workarounds
- use speedFactor to get a higher value of laneMaxV and compensate this in your stopSpeed,followSpeed functions
- use cfmodel vehicle variables to store a higher-than-laneMaxV speed in your stopSpeed/followSpeed functions and then apply this value in finalize speed

**Comment by Domsall** [comment]
The workaround at the moment that works perfectly for me is using  `v = maxV` instead of `double v = MIN2(maxV, laneMaxV)`.
I actually previously used a workaround similar to your first point, but that was kind of quick and dirty. I also tried to implement the second idea, but that doesn't work as intended here.

**Comment by namdre** [comment]
is this still a relevant issue?

**Comment by Domsall** [comment]
kind of. I changed my freeSpeed-function, so vehicles brake before reaching a new lane with a lower laneMaxV. Therefore the model can be used with the `MIN2(maxV, laneMaxV)` term.
But before that I used `v = maxV`, because vehicles with the Improved Intelligent Driver Model would brake a lot smoother (and I think a bit more realistic). Unfortunately they do not reach `laneMaxV` before getting to the new lane and then brake hard with the above term `MIN2(maxV, laneMaxV)`.
